### ISSUE TITLE
No output model after training globally

### ISSUE THREAD

**Issue by Daducica** [tensorflow/models#1295]
[issue happens when creating models with syntaxnet]

I'm trying to train a Syntaxnet model using [this tutorial](https://github.com/tensorflow/models/blob/master/syntaxnet/g3doc/syntaxnet-tutorial.md). I concatenated all the codes that are given there into one file, and I tried to run it.

The first two steps (i.e. training and evaluating the POS tagger, and training the parser locally) produce seemingly correct output files, yet when I try to run this code for global training, I don't get any output model files or error messages:

> bazel-bin/syntaxnet/parser_trainer \
>   --arg_prefix=brain_parser \
>   --batch_size=8 \
>   --decay_steps=100 \
>   --graph_builder=structured \
>   --hidden_layer_sizes=200,200 \
>   --learning_rate=0.02 \
>   --momentum=0.9 \
>   --output_path=models \
>   --task_context=models/brain_parser/greedy/$PARAMS/context \
>   --seed=0 \
>   --training_corpus=projectivized-training-corpus \
>   --tuning_corpus=tagged-tuning-corpus \
>   --params=200x200-0.02-100-0.9-0 \
>   --pretrained_params=models/brain_parser/greedy/$PARAMS/model \
>   --pretrained_params_names=\
> embedding_matrix_0,embedding_matrix_1,embedding_matrix_2,\
> bias_0,weights_0,bias_1,weights_1

The only error messages I get are about the nonexisting model files of global training, which should be in the "structured" folder but they aren't. Instead, there are only two files in the folder: "context" and "graph".

My issue is similar to [this one](https://github.com/tensorflow/models/issues/436). The difference is that  I have the problem only when trying to train globally. Also, I used a small dataset that I obtained from [this site](http://universaldependencies.org).

The full error/information message I get:
[Error message.txt](https://github.com/tensorflow/models/files/892994/Error.message.txt)

**Comment by calberti** [comment]
The error seems to be:
  Failed to find any matching files for models/brain_parser/structured/200x200-0.02-100-0.9-0/model

Is that path actually correct relative to where you're running bazel-bin/syntaxnet/parser_trainer from?  You should be able to fix this by changing --pretrained_params.

**Comment by Daducica** [comment]
When I'm training globally, it creates the folder "structured" and the subfolder "200x200-0.02-100-0.9-0". It even puts two files in it in the early stages of the training. So the path should be right.

This is where the paser_trainer is:
/common/tools/syntaxnet/models/syntaxnet/syntaxnet

And this is the root folder for all the models:
/common/tools/syntaxnet/models/syntaxnet/models
  

```
                             models
                        /              \
               brain_pos                brain_parser
                   /                    /            \ 
            greedy               greedy             structured
               /                    |                     \
128-0.08-3600-0.9-0     200x200-0.08-4400-0.85-4     200x200-0.02-100-0.9-0
        |                           |                         |
   models and               models and parsed             two files:
 tagged corpuses         corpuses that are put          context and graph
that are put there         there by the parser
  by the tagger
```

**Comment by Daducica** [comment]
So I took a look at the python files and I noticed that, in the case of global training, the condition 
`if num_steps % FLAGS.checkpoint_every == 0:`
returns false every time it's evaluated, meaning num_steps never reaches 5000.

Does anyone know why this happens?

Anyway, I replaced 5000 by 500, which seemed to solve my problem, but I'm not sure if I mess up something else by doing this.

**Comment by karmel** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
